ext/OceananigansNCDatasetsExt.jl

```diff
@@ -104,6 +104,13 @@ function squeezed_data(fd::AbstractField; array_type=Array{eltype(fd)}, with_hal
     return getindex(field_data_cpu, indices...)
 end
 
+function squeezed_data(fd::AbstractField; with_halos=false, kwargs...)
+    field_data = with_halos ? parent(fd) : interior(fd)
```

Comment:
does `construct_output` work here?

https://github.com/CliMA/Oceananigans.jl/blob/d3f0df86233828fb7c93b67591690c77102c3479/src/OutputWriters/output_construction.jl#L56

Reply:
No, they do different things. `squeeze_data()` should be specific to `NetCDFWriter` since it modifies the shape of the underlying array.

Reply:
Hmm ok. The only thing is that interpreting `with_halos` as the difference between `parent` and `interior` is different from what `construct_output` does? In particular the jld2 output writer uses the function `restrict_to_interior`:

https://github.com/CliMA/Oceananigans.jl/blob/d3f0df86233828fb7c93b67591690c77102c3479/src/OutputWriters/output_construction.jl#L11

Which I believe is more general than `interior` but I could be wrong.

Reply:
`construct_output()` is already used before squeeze is called, so by the time we reach `squeeze_data()` the user field `fd` should already have the proper indices. So I guess maybe `interior()` isn't needed? (In which case `with_halos` also isn't needed.) I'll try to make this change and see if tests pass.

Reply:
Ah, okay, the difference between `interior` and `parent` is indeed unnecessary when writing an output through `NetCDFWriter` because `construct_output()` was already called on the output. However, it matter when writing a `Field` directly into a `NetCDF` file.

That said, you be right that I might be able to simplify the logic here and re-use code from `output_construction.jl` to decrease the amount of lines. I'll try doing that.

Reply:
Tests are passing! I'm going to mark this as solved

Reply:
great work!

Reply:
Thanks! Can you approve this PR if there's no other comments? I really wanna start working on https://github.com/CliMA/Oceananigans.jl/pull/4848 again!